src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -41,7 +43,7 @@ def open_jar(self, mode):
 
     @property
     def _id(self):
-        return (self.target, self.analysis_file, self.classes_dir.path)
+        return (self.target, self.analysis_file, self.classes_dir.path, self.diagnostics_out)
```

Comment:
If I understand correctly, this means that the diagnostics path will end up in the cache key (but not the contents), right?